txpool/queue.go

```diff
@@ -47,7 +47,10 @@ func (q *accountQueue) unlock() {
 
 // prune removes all transactions from the queue
 // with nonce lower than given.
-func (q *accountQueue) prune(nonce uint64) (pruned []*types.Transaction) {
+func (q *accountQueue) prune(nonce uint64) (
+	pruned []*types.Transaction,
+	prunedHashes []types.Hash,
+) {
```

Comment:
Maybe have the hash collecting as a helper method defined elsewhere ? (just to keep code minimal)

Reply:
Why would we do the additional hash collection logic apart from this main loop?

Reply:
It's not intrinsically tied to the `accountQueue` - it's just collecting hashes from a slice of transactions.

Reply:
Yes - but the alternative is to create 2 separate loops for the pruned event hashes outside the `prune` method:
* One for the enqueued txs
* One for the promoted txs